romanisim/tests/test_linear.py

```diff
@@ -0,0 +1,144 @@
+"""Unit tests for linear module.
+
+Routines tested:
+- repair_coefficients
+- evaluate_nl_polynomial
+- apply
+"""
+import os
+import pytest
+import numpy as np
+from astropy import units as u
+import crds
+
+import roman_datamodels
+from romanisim import nonlinearity, parameters
+
+
+tijlist = [
+    [[1], [2, 3], [4, 5], [6, 7]],
+    [[100], [101, 102, 103], [110]],
+    [[1], [2], [3, 4, 5, 100]],
+]
+
+ma_table_list = [
+    [[1, 10], [11, 1], [12, 10], [30, 1], [40, 5], [50, 100]],
+    [[1, 1]],
+    [[1, 1], [10, 1]],
+]
+
+
+def test_linear_apply():
+    counts = np.random.poisson(100, size=(100, 100))
+    coeffs = np.asfarray([1.0, 0.7, 3.0e-6, 5.0e-12])
```

Comment:
I feel like the first coeff should almost always be close to zero.  Is that not the case in the real linearity files?  This is the ~pedestal / y-intercept, right?  I mean, it's fine to make it whatever, but it looks weird with a pedestal.

Reply:
I stole these numbers from STCAL, so it is possible they were made with JWST in mind? I didn't care too strongly about the numbers themselves, just the the orders of magnitude and that each number was different. I can change these to something else, if you want?

Reply:
It just seems to me like we should pick numbers that "look like" linearity coefficients.  Here's an image of the log10 of the absolute value some of the linearity coefficients
![image](https://github.com/spacetelescope/romanisim/assets/1417841/a3fe57c4-c049-4255-ad10-9d39e235c4f3)
The first coefficient is always zero and gets NaNed to white here.  The second coefficient is always very close to one.  Higher coefficients drop off very quickly (10^-5, 10^-10, 10^-15, ...).  It's of course okay to choose any set of coefficients, but when I read them I get the impression that you think that the first coefficient corresponds to the linear term rather than the constant term (since it is the linear terms in the reference file that are close to 1).  It does look to me like the stcal code follows the same convention (just a polynomial, with coefficients starting from a constant going to x^(ncoeff - 1)), so I'm surprised that the stcal tests would use these numbers.  But I don't think we need to copy them.

Reply:
Changed.